rfcs/0001-masonry-backend.md

```diff
@@ -0,0 +1,186 @@
+
+### Separating at the crate level vs the module level
+
+Masonry and Xilem's code could simply be merged in a single crate, under different modules.
+
+The consensus was mostly that having Masonry as a separate crate would help keep Xilem "honest" and enforce separation of concerns.
+
+More importantly, Masonry was built as a potential foundation for *multiple* native GUI crates, and it would be valuable to conserve that; therefore, while Masonry would still depend on crates from the Linebender ecosystem, it shouldn't be too tied to Xilem.
+
+### Porting Xilem to Masonry vs Masonry to Xilem
+
+In any merge between two projects, one has to be the base the other is grafted on.
+
+This RFC was written under the assumption that Masonry would be mostly unchanged and Xilem would be reworked to accomodate it, but it would also be possible to take changes from Masonry and graft them to Xilem's current backend code.
+
+That said, the Motivation section explains why Masonry is a healthier codebase to start from.
+
+## Unresolved questions
+
+### Do we want to keep the Masonry brand?
+
+The name Masonry was chosen when that crate wasn't officially part of the Linebender project. Even if we want the codebase and the crate to be added to the Linebender umbrella, its name and general branding are up in the air.
+
+Arguments for:
+
+- "Masonry" is a catchy name, easy to pronounce in most european languages.
+- The name already exists and the Rust community is vaguely aware of it, to the point is comes up (rarely) in discussions.
+- The name evokes brickwork and building stuff out of sturdy rectangle things. It can evoke having a solid foundation that things are built on top of.
+- I have [the coolest logo idea ever](https://github.com/PoignardAzur/masonry-rs/issues/7).
+
+Arguments against:
+
+- "A masonry layout" is a thing, especially in web development.
+  - People unfamiliar with Rust might come upon this crate and misunderstand what it's for.
+  - When talking about how Masonry does layout, discussion could get confusing.
+
+
+## Future possibilities
+
+### Performance features
+
+Xilem has added features over Druid that Masonry doesn't support, notably virtual lists and async support.
+
+I think they're currently under-used in Xilem, so Masonry doesn't need to implement them right away for parity.
+
+In the long term, these features will be very important for eventual performance work.
+
+### Directory structure
+
+Xilem's current directory structure is a bit awkward. It looks like:
+
+- `crates/`
+  - `xilem_core/`
+  - `xilem_web/`
+  - (soon) `masonry/`
+- `src/`
+
+In other words, all crates are in a `crates/` directory, except for `xilem/` which is in the root. This makes it seem like eg `xilem_web` is a dependency of `xilem`, when they're disjoint libraries.
```

Comment:
We could also move everything to `crates/` (so `crates/xilem`) and make it a pure workspace. Though I don't think the current structure is too bad, especially given that `xilem_web` is still kinda experimental (and either diverges fully, in case a separate repo may make sense, or can be hopefully merged in one way or the other at some point)

Relevant: https://xi.zulipchat.com/#narrow/stream/354396-xilem/topic/Moving.20.28or.20splitting.29.20the.20xilem.20crate.3F